test/e2e/config/e2e_conf.yaml

```diff
@@ -65,19 +65,19 @@ providers:
   - name: ${CAPI_FROM_RELEASE} # latest published release in the v1alpha4 series; this is used for v1alpha4 --> v1beta1 clusterctl upgrades test only.
     value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_FROM_RELEASE}/control-plane-components.yaml"
     type: "url"
-    contract: v1alpha4
+    contract: ${CONTRACT_FROM}
     replacements:
       - old: --metrics-addr=127.0.0.1:8080
         new: --metrics-addr=:8080
     files:
-      - sourcePath: "../data/shared/v1alpha4/metadata.yaml"
+      - sourcePath: "../data/shared/${CAPI_FROM_RELEASE:0:4}/metadata.yaml"
   - name: ${CAPI_TO_RELEASE}
     # Use manifest from source files
     value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_TO_RELEASE}/control-plane-components.yaml"
     type: "url"
-    contract: v1beta1
+    contract: ${CONTRACT_TO}
     files:
-      - sourcePath: "../data/shared/v1beta1/metadata.yaml"
+      - sourcePath: "../data/shared/${CAPI_TO_RELEASE:0:4}/metadata.yaml"
```

Comment:
is this variable naming convention correct ${CAPI_TO_RELEASE:0:4} ?  or is the collon doing some bash trick? where is the `v`?

Reply:
This just extracting sub-string